egs/swc/s5b/run.sh

```diff
@@ -0,0 +1,286 @@
+#!/bin/bash
+
+# This Kaldi recipe based on all three recordings in the Sheffield Wargame 
+# Corpora (SWC) is forked from the Kaldi recipe "s5b" for AMI data.
+# 
+# The interface for data downloading is NOT provided because that could not 
+# be automated for ethical concerns. Please visit the following website for 
+# details about the access to corpus data and language model:
+#
+#	http://mini-vm20.dcs.shef.ac.uk/swc/corpora_data.html
+#
+# Yulan Liu, 19 Sep 2016	
+
+
+. ./cmd.sh
+. ./path.sh
+
+echo "Starting time:"
+date
+
+# You may set 'mic' to:
+#  ihm [individual headset mic- the default which gives best results]
+#  sdm1 [single distant microphone- the current script allows you only to select
+#        the 1st of 8 microphones]
+#  mdm8 [multiple distant microphones-- currently we only support averaging over
+#       the 8 source microphones].
+# ... by calling this script as, for example,
+# ./run.sh --mic sdm1
+# ./run.sh --mic mdm8
+mic=ihm
+MODE='SA1'      # Other options: 'SA2', 'AD1', 'AD2'
+fastrun=1	# 1: skip unnecessary decoding 
+spkadapt=1	# 1: include speaker adaptation
+nj=20 		# defalut number of parallel jobs,
+stage=0
+
+. utils/parse_options.sh
+
+base_mic=$(echo $mic | sed 's/[0-9]//g') # sdm, ihm or mdm
+nmics=$(echo $mic | sed 's/[a-z]//g') # e.g. 8 for mdm8.
+
+set -euo pipefail
+
+# Path where SWC gets downloaded (or where locally available):
+SWCDIR=$PWD/wav_db # Default, 
+case $(hostname -d) in
+  minigrid.dcs.shef.ac.uk) SWCDIR=/share/spandh.ami1/usr/yulan/sw/kaldi/dev/kaldi-trunk/egs/swc/swc ;; # UoS
+esac
+
+
+# Check microphone type
+if [ "$base_mic" == "ihm" ] ; then		# Individual headset microphone
```

Comment:
This logic seems to be a bit too complicated... Also, can it be merged with the 'stage 2' ? What is the purpose of 'mic_ch' and 'PROCESSED_DIR'? Is it used only in the data preparation? You could also explain the role of the variables by comments...